x-pack/test/functional/services/ml/job_wizard_common.ts

```diff
@@ -619,5 +619,86 @@ export function MachineLearningJobWizardCommonProvider(
       });
     },
+
+    async togglingModelChangeAnnotationsShowsCalloutAndRemovesCallout() {
+      await mlCommonUI.toggleSwitchIfNeeded('mlJobWizardSwitchAnnotations', false);
+      await testSubjects.existOrFail('mlJobWizardAlsoEnableAnnotationsRecommendationCallout', {
+        timeout: 3_000,
+      });
+      await mlCommonUI.toggleSwitchIfNeeded('mlJobWizardSwitchAnnotations', true);
+      await testSubjects.missingOrFail('mlJobWizardAlsoEnableAnnotationsRecommendationCallout', {
+        timeout: 3_000,
+      });
+    },
+
+    async goToTimeRangeStep() {
+      await testSubjects.existOrFail('mlJobWizardTimeRangeStep', {
+        timeout: 3_000,
+      });
+      await testSubjects.click('mlJobWizardTimeRangeStep');
+      await testSubjects.existOrFail('mlJobWizardStepTitleTimeRange', {
+        timeout: 3_000,
+      });
+    },
+
+    async assertShortDurationTimeRange() {
+      const { startDate: origStartDate } = await this.getSelectedDateRange();
+
+      // calculate the new end datedate
+      const shortDurationEndDate = `${origStartDate?.split(':', 1)[0]}:01:00.000`;
+
+      // set the new end date
+      await testSubjects.setValue('mlJobWizardDatePickerRangeEndDate', shortDurationEndDate, {
+        clearWithKeyboard: true,
+        typeCharByChar: true,
+      });
+
+      // click away from time popover
+      await this.goToTimeRangeStep();
+
+      // assert time is set as expected
+      await this.assertDateRangeSelection(origStartDate as string, shortDurationEndDate);
+    },
+
+    async goToJobDetailsStep() {
+      await testSubjects.existOrFail('mlJobWizardJobDetailsStep', {
+        timeout: 3_000,
+      });
+      await testSubjects.click('mlJobWizardJobDetailsStep');
+      await testSubjects.existOrFail('mlJobWizardStepTitleJobDetails', {
+        timeout: 3_000,
+      });
+    },
+
+    async assertValidationCallouts(expectedCallOutSelectors: string[]) {
+      for await (const sel of expectedCallOutSelectors)
+        await testSubjects.existOrFail(sel, {
+          timeout: 3_000,
+        });
+    },
+
+    async assertCalloutText(calloutStatusTestSubj: string, expectedText: RegExp) {
+      const allCalloutStatusTexts = await testSubjects.getVisibleTextAll(calloutStatusTestSubj);
+
+      const oneCalloutMatches = allCalloutStatusTexts.some(
+        (visibleText) => !!visibleText.match(expectedText)
+      );
+      expect(oneCalloutMatches).to.eql(
+        true,
+        `Expect one of the callouts [${calloutStatusTestSubj}] to match [${expectedText}], instead found ${JSON.stringify(
+          allCalloutStatusTexts,
+          null,
+          2
+        )}`
+      );
+    },
+
+    async cloneJob() {
+      await testSubjects.click('euiCollapsedItemActionsButton');
+      await testSubjects.click('mlActionButtonCloneJob');
+      await testSubjects.existOrFail('mlPageJobWizardHeader-single_metric', {
+        timeout: 3_000,
+      });
+    },
```

Comment:
```suggestion
```
We already have a method to invoke the clone action https://github.com/elastic/kibana/blob/e4869071812028dae8af0b90ddee48b31d3a8dbf/x-pack/test/functional/services/ml/job_table.ts#L474

Reply:
https://github.com/elastic/kibana/commit/9ac54489cdb4848def0add2a53e39a3fb3589a40